src/main/java/org/codehaus/groovy/transform/ASTTransformationVisitor.java

```diff
@@ -221,7 +222,9 @@ private static void doAddGlobalTransforms(ASTTransformationsContext context, boo
                 String className;
                 BufferedReader svcIn = null;
                 try {
-                    svcIn = new BufferedReader(new InputStreamReader(service.openStream(), "UTF-8"));
+                    URLConnection urlConnection = service.openConnection();
+                    urlConnection.setUseCaches(false);
+                    svcIn = new BufferedReader(new InputStreamReader(urlConnection.getInputStream(), "UTF-8"));
```

Comment:
Readers will close their underlying Readers(e.g. `BufferedReader`(i.e. `svcIn`) --closes--> `InputStreamReader` --closes--> the stream instance created by `service.openStream()`), and `svcIn` will be closed in the `finally` block, could you please explain how file descriptor leaks?

Reply:
As explained in the commit message, the jar will be kept open by the cache in the jvm unless you deactivate that cache. This is meant as a performance optimization and unfortunately it's on by default.